### PR TITLE
Add German Translations

### PR DESCRIPTION
**Pull request by InterplanetaryEngineer**
Translated the pages Index, Learn, Getting Started, Overview, Samples, Tools, Why Zig, and ZSF.

Some of the translations may be a bit too literal, but hopefully not grammatically wrong; this took a bit longer than I expected, and I couldn't be bothered to rephrase every sentence into the most elegant prose.

The reader is addressed in the informal form, which I found most fitting for an open-source project talking to its community, but one might also argue that the formal pronouns are more appropriate for an organization addressing potential sponsors. I would like to hear another German speaker's opinion on this.

I have converted most section references from absolute to relative links (`[...](https://ziglang.org/#Stack-traces-on-all-targets)` -> `[...](#stacktraces-auf-allen-targets)`), but may have overlooked one or two.

### CODE REVIEW

**Comment by kristoff-it** [comment]
Thank you very much! I've pinged @MasterQ32 to give a read to the German phrasing. I'll soon do a pass to check links.

File: content/_index.de.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+---
+title: Home
+mobile_menu_title: "Home"
+---
+{{< slogan get_started="EINFÜHRUNG" docs="Dokumentation" notes="Änderungen">}}
```
Comment (kristoff-it) [inline content/_index.de.md:5]:
```suggestion
{{< slogan get_started="EINFÜHRUNG" docs="Dokumentation" notes="Änderungen" lang="de">}}
```

File: content/learn/getting-started.de.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+```
+ausgibt. Glückwunsch, du hast jetzt eine funktionierende Installation von Zig!
+
+## Nächste Schritte
+**Sieh dir auch auch die anderen Ressourcen im Abschnitt [Lernen](../) an**, suche dir die Dokumentation für deine Version (Hinweis: Nightly Builds sollten die `master`-Dokumenatation benutzen) und lies vielleicht [ziglearn.org](https://ziglearn.org).
+
+Zig ist ein junges Projekt, und wir haben leider nicht die Kapazitäten, umfangreiche Dokumentation und Lernmaterialien zu erzeugen. Wenn du feststeckst, kannst du aber auch [einer Zig-Community beitreten](https://github.com/ziglang/zig/wiki/Community), oder Initiativen wie [Zig SHOWTIME](https://zig.show) ansehen.
+
+Wenn dir Zig gefällt und du helfen willst, die Entwicklung voranzutreiben, kannst du [der Zig Software Foundation spenden](../../zsf).
+<img src="../../heart.svg" style="vertical-align:middle; margin-right: 5px">.
```
Comment (kristoff-it) [inline content/learn/getting-started.de.md:139]:
```suggestion
<img src="/heart.svg" style="vertical-align:middle; margin-right: 5px">.
```

File: content/zsf/index.de.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Unternehmenssponsoren
+
+### Geldspenden
+Die folgenden Unternehmen bieten der Zig Software Foundation direkte finanzielle Unterstützung, indem sie mehr als $1000 monatlich spenden.
+
+{{% sponsor-logos "monetary" %}}
+ <a href="https://pex.com" rel="noopener nofollow" target="_blank"><picture>
+   <picture>
+     <source srcset="../pex-white.svg" media="(prefers-color-scheme: dark)">
+     <img src="../pex-dark.svg">
```
Comment (kristoff-it) [inline content/zsf/index.de.md:52]:
```suggestion
     <source srcset="/pex-white.svg" media="(prefers-color-scheme: dark)">
     <img src="/pex-dark.svg">
```

File: content/zsf/index.de.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ </a>
+{{% /sponsor-logos %}}
+
+### Infrastruktur
+Die folgenden Unternehmen stellen ihre Dienste der Zig Software Foundation kostenfrei zur Verfügung.
+
+{{% sponsor-logos "services" %}}
+![](../lavatech.png)
+![](../dropbox.png)
+![](../aws.png)
```
Comment (kristoff-it) [inline content/zsf/index.de.md:63]:
```suggestion
![](/lavatech.png)
![](/dropbox.png)
![](/aws.png)
```

**Review by kristoff-it — CHANGES_REQUESTED** [review]
Just minor changes, sorry I realize I left a couple of foot guns here and there.

File: content/learn/samples.de.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+---
+title: "Codebeispiele"
+mobile_menu_title: "Codebeispiele"
+toc: true
+---
+
+## Speicherlecks aufdecken
+Der `std.GeneralPurposeAllocator` kann Doppel-Freigabe und Speicherlecks nachverfolgen.
```
Comment (MasterQ32) [inline content/learn/samples.de.md:8]: `std.heap.GeneralPurposeAllocator` cc @kristoff-it

File: content/learn/why_zig_rust_d_cpp.de.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+Zigs Standardbibliothek bietet Features, die Heapallokationen bereitstellen und damit arbeiten, aber diese sind optional, nicht in die Sprache selbst eingebaut.
+Wenn du nie einen Allocator initialisierst, kannst du dir sicher sein, dass dein Programm nichts alloziiert.
+
+Jedes Feature der Standardbibliothek, das Heapspeicher alloziieren muss, nimmt dazu einen `Allocator` als Parameter an. Das bedeutet, dass Zigs Standardbibliothek Freestanding-Targets unterstützt. Zum Beispiel können `std.ArrayList` und `std.AutoHashMap` für Bare-Metal-Programme genutzt werden!
+
+Eigene Allocators machen manuelle Speicherverwaltung kinderleicht. Zig hat einen Debug-Allocator, der Speichersicherheit bei Use-after-free und Double-free garantiert. Er detektiert und logt automatisch Speicherlecks. Es gibt einen Arena-Allocator, der beliebig viele Allokationen bündeln und zusammen freigeben kann (statt sie einzeln zu verwalten). Besondere Allocators können genutzt werden, um Performance oder Speichernutzung nach den Anforderungen einer Anwendung zu optimieren.
```
Comment (MasterQ32) [inline content/learn/why_zig_rust_d_cpp.de.md:47]: Allokatoren, nicht Allocators

File: content/learn/why_zig_rust_d_cpp.de.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+* Derzeit ist aus pragmatischer Sicht C die vielseitigste und portabelste Sprache. Jede Sprache, die nicht mit mit C-Code interagieren kann, riskiert es, in Obskurität zu verschwinden. Zig versucht, die neue portable Sprache für Bibliotheken zu werden, indem es gleichzeitig C ABI-Konformität für externe Funktionen vereinfacht und Sicherheit und ein Sprachdesign einführt, das häufige Fehler innerhalb der Implementierungen verhindert.
+
+## Ein Paketmanager und Build-System für bestehende Projekte
+
+Zig ist eine Programmiersprache, aber sie bringt auch ein Build-System und einen Paketmanager mit, die auch für traditionelle C/C++-Projekte nützlich sein sollen.
+
+Zig kann nicht nur C- oder C++-Code ersetzen, sondern auch autotools, cmake, make, scons, ninja usw. Und obendrein bietet es (bald™) einen Paketmanager für native Abhängigkeiten. Dieses Build-System soll auch für Projekte relevant sein, deren gesamte Codebasis aus C oder C++ besteht.
```
Comment (MasterQ32) [inline content/learn/why_zig_rust_d_cpp.de.md:69]: Dieses Build-System ist auch für Projekte relevant, deren gesamte Codebasis aus C oder C++ besteht.

**Review by MasterQ32 — CHANGES_REQUESTED** [review]
Made most comments in german, there are mostly some formulations that need change (most time with an improved formulation).

If those are changed, we can go live with this :+1:

File: content/learn/getting-started.de.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+**<center>Hinweis für Nutzer von Apple Silicon</center>**
+Zigs Unterstützung für Code Signing ist experimentell. Du kannst Zig mit deinem M1 Mac vewenden,
+aber momentan ist der einzige Weg, Zig für arm64 macOS zu bekommen, es selbst zu kompilieren.
+Beachte dazu den Abschnitt [Vom Quellcode kompilieren](#vom-quellcode-kompilieren).
+{{% /div %}}
+
+
+## Release oder Nightly Build?
+Zig hat noch nicht v1.0 erreicht und der aktuelle Release-Zyklus ist an neue Releases von LLVM gebunden, die etwa alle 6 Monate erscheinen.
+In der Praxis **liegen Zigs Releases weit auseinander und werden bei der momentanen Entwicklungsgeschwindigkeit irgendwann veraltet**.
```
[the remark this answers is not in the document]
Comment (InterplanetaryEngineer) [reply content/learn/getting-started.de.md:17]: "irgendwann" is the faithful translation of the "eventually" on the English page. Is that one also incorrect?

File: content/learn/getting-started.de.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+Die getaggten Versionen können verwendet werden, aber wenn du tiefer in Zig einsteigen möchtest, **empfehlen wir, auf ein Nightly Build umzusteigen**, vor allem, weil du so einfacher Hilfe bekommen kannst: der Großteil der Community und Seiten wie 
+[ziglearn.org](https://ziglearn.org) verfolgen aus obigen Gründen den Master-Branch.
```
[the remark this answers is not in the document]
Comment (InterplanetaryEngineer) [reply content/learn/getting-started.de.md:20]: The German (Git manual)[https://git-scm.com/book/de/v2/Git-Branching-Remote-Branches] also translates "track" as "vefolgen"

File: content/learn/_index.de.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+---
+
+# Lernen
+Dieser Abschnitt enthält Ressourcen, die hoffentlich helfen, Zigs Philosophie zu verstehen.
+
+## Dokumentation
+{{% learn_docs %}}
+
+## Einführung
+Dies sind Einführungen in Zig für Programmierer mit verschiedenen Vorkenntnissen.
```
[the remark this answers is not in the document]
Comment (InterplanetaryEngineer) [reply content/learn/_index.de.md]: Der Satz ist so grammatikalisch in Ordnung, das "alles" ist redundant, und "Hier sind" hört sich ungelenk an...

File: content/learn/getting-started.de.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (MasterQ32) [reply content/learn/getting-started.de.md:17]: Yeah, but i was going less for a word-by-word translation and brought more intent into the sentence. @kristoff-it *eventually* might actually be "too slow" for a time span of 1 to 6 weeks when releases are 6 months apart

File: content/learn/getting-started.de.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (MasterQ32) [reply content/learn/getting-started.de.md:20]: Yeah, i know. But "verfolgen" just sounds weird and is less clear imho

File: content/learn/_index.de.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (MasterQ32) [reply content/learn/_index.de.md]: Da, aber da die Seite informell formuliert ist und nicht siezt, finde ich einen etwas lockerern Sprachstil angebrachter. Das ist aber denke ich Geschmackssache

[the remark this answers is not in the document]
Comment (InterplanetaryEngineer) [reply content/learn/_index.de.md]: Das stimmt allerdings. Ist die letzte Fassung:

> Einige Seiten zum Einstieg in Zig [...]:

so okay?

**Review by MasterQ32 — COMMENTED** [review]
It's good now from my side.

**Review by MasterQ32 — APPROVED** [review]

**Comment by kristoff-it** [comment]
Thank you very much!

**Pull request merged**